src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -95,28 +96,47 @@ private void OnProjectsChanged(ConnectionInformation connection, ProjectInformat
 
         #region Workflow steps
 
+        private static void AbortWorkflow(IProgressController controller, CancellationToken token)
+        {
+            bool aborted = controller.TryAbort();
+            Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+        }
+
         internal /* for testing purposes */ void ConnectionStep(IProgressController controller, CancellationToken cancellationToken, ConnectionInformation connection, IProgressStepExecutionEvents notifications)
         {
             this.owner.UserNotification?.HideNotification(NotificationIds.FailedToConnectId);
+            this.owner.UserNotification?.HideNotification(NotificationIds.BadServerPluginId);
 
             notifications.ProgressChanged(connection.ServerUri.ToString(), double.NaN);
 
             ProjectInformation[] projects = this.owner.SonarQubeService.Connect(connection, cancellationToken)?.ToArray();
```

Comment:
@eyalantzMS Scratch that last comment.
Can't detect the different cases where we have a connection failure (auth, bad server name) vs not having any version of the C# plugin installed.
In both cases the `GetPluginVersion` API will return `null`.